pennylane/_device.py

```diff
@@ -318,7 +318,7 @@ def check_validity(self, queue, observables):
                 to be evaluated on the device
         """
         for o in queue:
-            if o.name not in self.operations:
+            if o.name not in self.operations and not o.is_inverse:
```

Comment:
What is the idea behind this? Right now, if the device does not support `PauliZ` it would not raise an error on `PauliZ(0).inv()`.

Shouldn't there be more of a check that the device supports inversion at all, analogous to the tensor observables?

Reply:
That's a great point there, thank you!

I have modified the entire ``check_validity`` method, now it uses the ``supports_inverse``, ``supports_observable`` and ``supports_operation`` methods.

Added the test ``test_check_validity_with_not_supported_operation_inverse`` that attempts to catch the scenarios that you have described.